rfcs/text/0408-versioning-types.md

```diff
@@ -0,0 +1,344 @@
+        1.  The values are defined by either:
+            1.  An empty object (until Link Constraints are implemented)
+            1.  An array definition, where
+                1.  the items are an empty object (until Link Constraints are implemented)
+                1.  `ordered` is optionally defined
+                1.  `minItems` and/or `maxItems` are optionally defined
+    1.  `requiredLinks` array, where the elements are _versioned URIs_, and is a subset of the keys in the `links`
+
+An Entity Type `A` is therefore compatible with another Entity Type `B` if and only if
+
+1.  Each key in the `properties` of `A` is also in the `properties` of `B`
+1.  If a property in `A` is defined as an array, then it's also defined as an array in `B` (and therefore the same for direct references)
+1.  If a property in `A` is defined as an array with a `minItems`, then there is a `minItems` constraint on the respective property in `B` that is equal to or larger to that in `A`
+1.  If a property in `A` is defined as an array with a `maxItems`, then there is a `maxItems` constraint on the respective property in `B` that is less than or equal to that in `B`
+1.  Each property type referenced in `A` is compatible with the respective property type referenced in `B`. Where respective is defined by having the same base URI (property key)
+1.  Each URI in the `required` list of `A` is also in the `required` list of `B`
+1.  Each key in the `links` of `A` is also in the `links` of `B`
+1.  If a link in `links` of `A` is defined as an array, then it's also defined as an array in `B` (and therefore the same for direct references)
+1.  If a link in `links` of `A` is defined as an array with a `minItems`, then there is a `minItems` constraint on the respective link in `B` that is equal to or larger to that in `A`
+1.  If a link in `links` of `A` is defined as an array with a `maxItems`, then there is a `maxItems` constraint on the respective link in `B` that is less than or equal to that in `B`
+1.  If a link in `links` of `A` is defined as an array with `ordered` as `true`, then there is an `ordered` as `true` on the respective link in `B`
+1.  Each URI in the `requiredLinks` list of `A` is also in the `requiredLinks` list of `B`
+
+### Link Types
+
+Link Types are only able to be changed through _semantic annotations_ and as such, any Link Type `X`, with the same Base URI as another Link Type `Y` (i.e. a different version of the same link type), are _compatible_.
+
+## When to Check Compatibility of Types
+
+Block schemas, and other uses of types (for example graph service methods), can request data from the embedding application by referencing types. Not relying on the version to ensure compatibility means that if a block, or data (associated with a versioned type) in an embedding application, is "outdated", the compatibility can be checked by comparing the two schemas. Data blobs can continue to be validated against the relevant schema.
```

Comment:
```suggestion
Block schemas, and other uses of types (for example graph service methods), can request particular data structures from the embedding application by referencing types. Not relying on the version of a type to ensure compatibility means that if a block or data (associated with a versioned type) in an embedding application is associated with a different version of a type – or different types altogether  – the compatibility of the data can still be checked by comparing the two schemas. Data blobs can continue to be validated against the relevant schema.
```

Reply:
Have added, although I kept the commas as I think they are correct here?

Reply:
No harm keeping the commas, just thought they slowed it down a bit